### PR TITLE
MultiServer: SetReply `_read_hints_{team}_{slot}` update when item(s) found.

### PR DESCRIPTION
**Pull request by ThePhar**
## What is this fixing or adding?
Currently, subscribers to the `_read_hints_{team}_{slot}` data storage key do not get notified of an update to hints unless a new hint was added. So, for example, if it's displayed as `(not found)` in the CommonClient "Hints" tab (or other clients that watch for this key), it will not be notified to update when location has been found, and will erroneously continue to report location as still not found until a new hint is created. This has already caused confusion as seen [this discord bug report](https://discord.com/channels/731205301247803413/1178523651134476318).

This tweak rebuilds the hints when a `LocationChecks` package is received and if any newly checked locations already exist in the hints set for the relevant slots (meaning it was previous "not found", but now is), it will update their hints and send a `SetReply` package notifying them of the updated hints set.

## How was this tested?
Made a hint, then checked location and saw hints update in real time.

```
Incoming message: [{"cmd":"LocationChecks","locations":[69696968]}]
(Team #1) Player5 sent Button Activation to Player1 (The Item on the Desk)
Outgoing broadcast: [{"cmd":"PrintJSON","data":[{"text":"5","type":"player_id"},{"text":" sent "},{"text":"69696968","player":1,"flags":1,"type":"item_id"},{"text":" to "},{"text":"1","type":"player_id"},{"text":" ("},{"text":"69696968","player":5,"type":"location_id"},{"text":")"}],"type":"ItemSend","receiving":1,"item":{"item":69696968,"location":69696968,"player":5,"flags":1,"class":"NetworkItem"}}]
Outgoing broadcast: [{"cmd":"SetReply","key":"_read_hints_0_1","value":[{"receiving_player":1,"finding_player":5,"location":69696968,"item":69696968,"found":true,"entrance":"","item_flags":1,"class":"Hint"}]}]
Outgoing broadcast: [{"cmd":"SetReply","key":"_read_hints_0_5","value":[{"receiving_player":1,"finding_player":5,"location":69696968,"item":69696968,"found":true,"entrance":"","item_flags":1,"class":"Hint"}]}]
```

## If this makes graphical changes, please attach screenshots.

Item hinted from Player1 (left+TextClient) in Player5's (right) game.
image

After "collecting" item in Player5's game, TextClient has been updated.
image

### CODE REVIEW

**Comment by Jarno458** [comment]
Did you also check this with item_links? because by looking at the code i think those might have been forgotten

File: MultiServer.py [hunk 1]
```diff
@@ -955,18 +960,34 @@ def register_location_checks(ctx: Context, team: int, slot: int, locations: typi
     if new_locations:
         if count_activity:
             ctx.client_activity_timers[team, slot] = datetime.datetime.now(datetime.timezone.utc)
+
+        ctx.location_checks[team, slot] |= new_locations
+        slots_with_updated_hints: typing.Set[int] = set()
+        hint_locations_to_receiver: typing.Dict[int, int] = {
```
Comment (Berserker66) [inline MultiServer.py:966]: This looks like a very serious performance degredation in one of our hottest code paths.
If you run `ctx.get_rechecked_hints(team, slot)` anyway, why even any of the surrounding code.

Comment (ThePhar) [reply MultiServer.py:966]: I think ultimately, I may just change how hints are stored while in memory.

**Comment by ThePhar** [comment]
Moving to draft. Plan after work is to rework this.

**Pull request closed without merging**